### ISSUE TITLE
Default getExampleFilename implementation produces false positives

### ISSUE THREAD

**Issue by okonet** [styleguidist/react-styleguidist#1430]
**Current behavior**

Right now, setting `skipComponentsWithoutExample: true` and giving the following directory structure

![CleanShot 2019-09-10 at 15 05 27](https://user-images.githubusercontent.com/11071/64616282-8c341480-d3dc-11e9-8cc6-27388fe6ae29.png)

styleguidist will render `Form.md` contents twice: once for `Form.js` file and once for `Field.js`. For the first example, it will result in an error `TypeError: Form is not defined` for the `Field.js` one.

**To reproduce**

Create 2 components that start with the same prefix: `Form.js` and `Field.js` in this example and add `Form.md` to the directory.

**Expected behavior**

It should only render `Form` component once.

To work around I had to add this to the config:

```js
getExampleFilename(componentPath) {
    return componentPath.replace(/\.jsx?$/, '.md')
  },
```

**Comment by okonet** [comment]
@bebraw could you come up with a reproducible repo when you have some time perhaps?

**Comment by sapegin** [comment]
Yes, `Foldername/Foldername.md` is one of the patterns that the default function will try. So it works as expected. 

Rendering a component twice is another thing: your components pattern is likely too broad. Does it match `index.js` as well? This could be a reason.

**Comment by okonet** [comment]
No, it doesn’t afaik. It only matches components with first capital letter. It matches `Field` is matched but it doesn’t have the example file (`Field.md`) so it shouldn’t render.

**Comment by anton6** [comment]
Can this issue be re-opened please? Or do I need to open new one instead?

Facing the same problem at the moment.

I'm using index files as an import alias and grouping component related files in same folder (exactly as per OPs example).

I have set `skipComponentsWithoutExample` to `true` but it's still picking up **all** other files inside the component folder even the ones without the `md` files.

When I define the components match pattern it does not pick up the index file because my index has `.js` extension and all the files inside of the folder are `.jsx` extension, and the pattern I use is `.../**/*.jsx`.

So the original rule:

> Yes, Foldername/Foldername.md is one of the patterns that the default function will try. So it works as expected.

Doesn't cover this more complex use case unfortunately.

**Comment by anton6** [comment]
And if I use pattern `.../**/*.js` although styleguidist correctly renders only 1 component example there is no component props documented/displayed because the index file only contains component import and default export so somehow it's not able to figure out any further documentation.
